lib/util/git/repo-store.ts

```diff
@@ -0,0 +1,250 @@
+import { spawn } from 'child_process';
+import { GlobalConfig } from '../../config/global';
+import { logger } from '../../logger';
+import type { LongCommitSha } from './types';
+import { syncGit } from '.';
+
+interface ExecResult {
+  code: number | null;
+  stdout: string;
+  stderr: string;
+}
+
+function execGit(cmd: string[], input?: string): Promise<ExecResult> {
+  const cwd = GlobalConfig.get('localDir');
+
+  return new Promise((resolve, reject) => {
+    const proc = spawn('git', cmd, { cwd });
+
+    const stdout: string[] = [];
+    const stderr: string[] = [];
+
+    if (input) {
+      proc.stdin.write(input);
+      proc.stdin.end();
+    }
+
+    proc.stdout.on('data', (data) => {
+      stdout.push(data.toString());
+    });
+
+    proc.stderr.on('data', (data) => {
+      stderr.push(data.toString());
+    });
+
+    proc.on('close', (code) => {
+      resolve({
+        code,
+        stdout: stdout.join(''),
+        stderr: stderr.join(''),
+      });
+    });
+
+    proc.on('error', reject);
+  });
+}
+
+async function createBlob(content: string): Promise<LongCommitSha | null> {
+  const { code, stdout, stderr } = await execGit(
+    ['hash-object', '-w', '--stdin'],
+    content,
+  );
+
+  if (code !== 0) {
+    logger.debug(`git hash-object failed: ${stderr}`);
+    return null;
+  }
+
+  return stdout.trim() as LongCommitSha;
+}
+
+interface TreeEntry {
+  sha: string;
+  path: string;
+}
+
+async function createTree(entries: TreeEntry[]): Promise<LongCommitSha | null> {
+  const treeEntries = entries
+    .map(({ sha, path }) => `100644 blob ${sha}\t${path}`)
+    .join('\n');
+
+  const { code, stdout, stderr } = await execGit(['mktree'], treeEntries);
+
+  if (code !== 0) {
+    logger.debug(`git mktree failed: ${stderr}`);
+    return null;
+  }
+
+  return stdout.trim() as LongCommitSha;
+}
+
+interface CreateCommitOptions {
+  tree: LongCommitSha;
+  message: string;
+  parents?: LongCommitSha[];
+}
+
+async function createCommit({
```

Comment:
Is this an "orphan" commit we push?

Reply:
Yes, I'm not sure it's okay or not

Reply:
I wonder if it's better to create an orphan commit, with no parent, or to make it branch off the very first commit of the repo ever